pkg/auth/dependency/provider/password/password.go

```diff
@@ -8,6 +8,6 @@ type Provider interface {
 	CreatePrincipal(principal Principal) error
 	GetPrincipalByLoginIDWithRealm(loginIDKey string, loginID string, realm string, principal *Principal) (err error)
 	GetPrincipalsByUserID(userID string) ([]*Principal, error)
-	GetPrincipalsByEmail(email string) ([]*Principal, error)
+	GetPrincipalsByLoginID(loginIDKey string, loginID string) ([]*Principal, error)
```

Comment:
login id key is not part of the unique constraint `(login_id, realm)`. This method should not take `loginIDKey`. Similarly, GetPrincipalByLoginIDWithRealm should not take `loginIDKey`

Reply:
Login ID may be normalize based on the login ID key. I don't want to do it in handler, so it should be in this place. Later may divide this interface into Provider and Store, so `GetPrincipalsByLoginID` has login ID key parameter in Provider, and no login ID key parameter in Store.